packages/react/src/blocks/CustomCode.tsx

```diff
@@ -90,6 +90,10 @@ class CustomCodeComponent extends React.Component<Props> {
     }
   }
 
+  shouldComponentUpdate(nextProps: Readonly<Props>): boolean {
+    return nextProps.code !== this.props.code;
+  }
```

Comment:
we should be able to test this with a content entry that has a custom code block that increments a `state.foo`. when the bug is present, it should update it twice, and when the fix is there it should update once.

Reply:
i was only able to reproduce this issue when I was using gen1 React SDK in a next.js app directory project. If we use pages directory or even vanilla react page orientation doesn't trigger a re-render